File: ConflictsWeb/src/models/RegisterRequest.ts
export class RegistrationRequest{
    public username: string;
    public password: string;
    public passwordConfirm: string;

    constructor(username: string, password: string, passwordConfirm: string) {
        this.username = username;
        this.password = password;
        this.passwordConfirm = passwordConfirm;
    }
}

File: ConflictsServer/src/main/java/islab/project/conflictsserver/auth/RegisterRequest.java
package islab.project.conflictsserver.auth;

import islab.project.conflictsserver.validators.UniqueUsername;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest {

    @NotEmpty
    @UniqueUsername
    private String username;
    @NotEmpty
    private String password;

}
